### PR TITLE
CLIMATE-634 please take a look at this request as a reference.

### PR DESCRIPTION
**Pull request by huikyole**
1. Currently, we cannot load ocw.dataset or ocw.datset_processor library in utils.py. So calc_climatology_season is moved from ocw.utils to ocw_dataset_processor
2. A new module, temporal_subset is added.
3. A new example code to test 'calc_climatology_season' is added. 

I hope this helps.

### CODE REVIEW

File: examples/simple_model_to_model_bias_DJF_and_JJA.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+print("Loading %s into an OCW Dataset Object" % (FILE_1_PATH,))
+knmi_dataset = local.load_file(FILE_1_PATH, "tasmax")
+print("KNMI_Dataset.values shape: (times, lats, lons) - %s \n" % (knmi_dataset.values.shape,))
+
+print("Loading %s into an OCW Dataset Object" % (FILE_2_PATH,))
+wrf_dataset = local.load_file(FILE_2_PATH, "tasmax")
+print("WRF_Dataset.values shape: (times, lats, lons) - %s \n" % (wrf_dataset.values.shape,))
+
+""" Step 2: Calculate seasonal average """
+print("Calculate seasonal average")
```
Comment (kwhitehall) [inline examples/simple_model_to_model_bias_DJF_and_JJA.py]: In this example, to replicate the bug associated with the CLIMATE-634 tix, do below:

knmi_dataset.values, knmi_DJF = utils.calc_climatology_season(month_start=12, month_end=2, dataset=knmi_dataset)
wrf_dataset.values, wrf_DJF = utils.calc_climatology_season(month_start=12, month_end=2, dataset=wrf_dataset)
print("Seasonally averaged KNMI_Dataset.values shape: (times, lats, lons) - %s \n" % (knmi_dataset.values.shape,))
print("Seasonally averaged wrf_Dataset.values shape: (times, lats, lons) - %s \n" % (wrf_dataset.values.shape,))
#JJA works with the current functionality
# knmi_dataset.values, knmi_JJA = utils.calc_climatology_season(month_start=6, month_end=8, dataset=knmi_dataset)
# wrf_dataset.values,wrf_JJA = utils.calc_climatology_season(month_start=6, month_end=8, dataset=wrf_dataset)

**Comment by huikyole** [comment]
presentation1

**Comment by MJJoyce** [comment]
The problem with the movement of calc_climatology_season in the format it was moved in is that it doesn't follow the dataset processor conventions. A dataset_processor function takes an OCW Dataset and some parameters and returns a modified OCW dataset. If you want a function that takes an OCW Dataset and returns some other information then that should be in utils.

Also I'm not seeing how this slide relates to the other comments in this thread, but I think I'm missing something?

**Comment by huikyole** [comment]
Now in the latest commit, utils.calc_climatology_year, utils.calc_climatology_season and utils.calc_climatology_montly can be all replaced by using dataset_processor.temporal_subset and calc_temporal_mean. Also all of the issues in the old calc_climatology_season have been addressed.

**Pull request merged**